### ISSUE TITLE
Add VS Code problem matcher for watch mode

### ISSUE THREAD

**Issue by srmagura** [skovy/typed-scss-modules#169]
Thanks for this great tool!

## Current Behavior

A very common development workflow using `typed-scss-modules` would be to run `typed-scss-modules --watch` and then start up `tsc --watch` once the `.d.ts` files have been generated.

In VS Code, this would typically be accomplished via [Tasks](https://code.visualstudio.com/docs/editor/tasks). However, currently there is no way for VS Code to know when `typed-scss-modules --watch` has completed the initial compilation.

## Possible Solution

I am proposing we...

1. Add "compilation started" and "compilation completed" output messages to `typed-scss-modules --watch` so that a VS Code problem matcher can detect when compilation is in progress.
2. Create a VS Code problem matcher for `typed-scss-modules` that utilizes the "compilation started" and "compilation ended" messages to support the use case I described above. [This documentation shows how to accomplish this.](https://code.visualstudio.com/docs/editor/tasks#_background-watching-tasks)

## Implementation

I am considering adding `typed-scss-modules` to my company's build tooling, and, if we decide to commit to it, I would be happy to implement this feature.

Are you open to a PR that implements this? 

Starting out, I would like to include the JSON the defines the problem matcher in the README — then users can simply copy it into their codebase. We could distribute the problem matcher via a VS Code extension, though that is likely overkill for this use case.

**Comment by skovy** [comment]
Hey @srmagura, overall this solution sounds reasonable to me. To recap what you want to do:

- Verbose log (so it can be turned off with `--logLevel` option) that compilation started/ended
- Add an example to the README for the VSCode config JSON so people can copy/paste to their own projects

If that sounds right I think this sounds reasonable and like a great contribution if you want to open a PR.

**Comment by srmagura** [comment]
I tried implementing this but ultimately determined it would not be possible without non-trivial changes to typed-scss-modules' file watcher. Basically, typed-scss-modules handles each file individually when watching, but VS Code needs a single "compilation started" and a single "compilation completed" message (even though multiple files could have changed).

I was able to solve my use case by making two VS Code tasks. Task 1 runs `yarn typed-scss-modules` (non-watch mode) and Task 2 runs `yarn typed-scss-modules --watch --ignoreInitial`. This allows my other tasks (e.g. TypeScript compiler) to start once Task 1 is complete.

Thank you :)

**Comment by skovy** [comment]
Oh, that solution sounds clever and maybe should be considered the official solution? 🤔 

Do you still want to add the VSCode config for that to the README for others that may run into the same problem?

**Comment by srmagura** [comment]
Hmm, I actually stopped using that solution because I found it to be a bit annoying in practice 😂. The issue is that you will have two terminals running typed-scss-modules simultaneously if you rerun the main build task. So I don't think it is a good enough solution to add to the README, but maybe someone will stumble upon this issue if they encounter the same problem.

Now I've gone back to just having one task that runs typed-scss-modules in watch mode.

**Comment by skovy** [comment]
> Basically, typed-scss-modules handles each file individually when watching, but VS Code needs a single "compilation started"

Can we add a new event listener for `ready`? https://github.com/paulmillr/chokidar I haven't tested but could solve the "compilation started".

```
  .on('ready', () => alerts.info('Initial scan complete. Ready for changes'))
```

For "compilation completed," could that be added to the existing `change` event listener (or use the existing message), since in practice only one CSS file is commonly saved at a time? And the worst case if multiple CSS files change, the message is output a few extra times. For this use case, that would restart the TS compiler each time? That seems like expected behavior?

**Comment by srmagura** [comment]
Nice ideas 😃.

I actually had the "Initial compilation completed" message working when I tested this. The difficulty is printing "File change detected. Compiling..." when a file changes — without printing this message one time for each file that changed. I don't think the `ready` event would help here since it only occurs once, after the initial scan is complete.

> since in practice only one CSS file is commonly saved at a time?

It's not uncommon for me to save multiple Sass files at once with "save all", so I wouldn't want to assume only one file is changed at once.

> For "compilation completed," could that be added to the existing change event listener (or use the existing message),

My concern with this approach is that it would break down in the following situation. (This is actual output copied from my terminal by the way. I think this specific output occurred when I switched branches.)

```text
[ADDED] src/auth/CreateOrgForm.module.scss
[ADDED] src/auth/Goodbye.module.scss
[ADDED] src/auth/LoginLayout.module.scss
[ADDED] src/auth/Wizard.module.scss
[GENERATED TYPES] src/auth/CreateOrgForm.module.scss.d.ts
[GENERATED TYPES] src/auth/Goodbye.module.scss.d.ts
[GENERATED TYPES] src/auth/LoginLayout.module.scss.d.ts
[GENERATED TYPES] src/auth/Wizard.module.scss.d.ts
```

If we considered the compilation done the first time we see `[GENERATED TYPES]`, that would be premature since there are still more files we need to generate types for.

I considered an approach where we would increment a counter for each `[ADDED]` / `[CHANGED]` event, and then decrement the counter for each `[GENERATED TYPES]` event but that just seemed complex and error prone. 😞
